pkg/storage/chunk/cache/resultscache/interface.go

```diff
@@ -0,0 +1,56 @@
+package resultscache
+
+import (
+	"context"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+type Request interface {
+	proto.Message
+	// GetStart returns the start timestamp of the request in milliseconds.
+	GetStart() time.Time
+	// GetEnd returns the end timestamp of the request in milliseconds.
+	GetEnd() time.Time
+	// GetStep returns the step of the request in milliseconds.
+	GetStep() int64
+	// GetQuery returns the query of the request.
+	GetQuery() string
+	// GetCachingOptions returns the caching options.
+	GetCachingOptions() CachingOptions
+	// WithStartEnd clone the current request with different start and end timestamp.
+	WithStartEnd(start time.Time, end time.Time) Request
+}
+
+type Response interface {
+	proto.Message
+}
+
+// ResponseMerger is used by middlewares making multiple requests to merge back all responses into a single one.
+type ResponseMerger interface {
+	// MergeResponse merges responses from multiple requests into a single Response
+	MergeResponse(...Response) (Response, error)
+}
```

Comment:
I find only one implementation (`queryMergerAsCacheResponseMerger`) of the ResponseMerger interface. Is this really needed, since it only casts back and forth?